src/wstore/asset_manager/asset_manager.py

```diff
@@ -54,11 +59,67 @@ def _save_resource_file(self, provider, file_):
             file_name = file_.name
             file_.seek(0)
             content = file_.read()
-
         # Check file name
         if not is_valid_file(file_name):
             logger.debug(f"`{file_name}` is not a valid file name")
             raise ValueError("Invalid file name format: Unsupported character")
+        if isPublic is True and settings.AWS_ENABLED:
+            return self.__save_resource_aws(file_name, content)
+        else:
+            return self.__save_resource_local(provider, file_name, content)
+        
+
+    def __save_resource_aws(self,  file_name, content):
+
+        # Create temp dir for assets if it does not exist
+        provider_dir = os.path.join(settings.MEDIA_ROOT, "assets", "temp")
+
+        if not os.path.isdir(provider_dir):
+            os.makedirs(provider_dir, exist_ok=True)
+        # Local file path for the asset file 
+        file_path = os.path.join(provider_dir, file_name)
+        logger.debug(f"File path for asset file: {file_path}")
+        # Key of s3
+        resource_path = os.path.join(settings.MEDIA_DIR, file_name)
+
+        if resource_path.startswith("/"):
+            resource_path = resource_path[1:]
+
+        # Check if the file already exists
+        if os.path.exists(file_path):
+            logger.error(f"The asset file `{file_name}` already exists")
+            raise ConflictError(f"The provided digital asset file ({file_name}) already exists")
+
+        logger.debug("Paths needed for saving temp resource file OK")
+        
+        # Create file
+        with open(file_path, "wb") as f:
+            f.write(content)
+
+        self.rollback_logger["files"].append(file_path)
+
+        logger.debug("Asset temp file created")
+        # upload the file to S3
+        acl = {'ACL': 'public-read'} if settings.ACL_ENABLED else None
+        self.s3.upload_file(file_path, settings.BUCKET_NAME, resource_path, ExtraArgs=acl)
+
+        logger.debug(f'The file {file_name} is uploaded to {resource_path} in s3')
+
+        # delete file in local
+        try:
+            logger.debug(f"Deleting file {file_path}")
+            os.remove(file_path)
+            logger.debug(f"Deleting file {file_path}2")
+        except:
+            raise ConflictError(f"The provided digital asset file ({file_name}) cannot be removed")
+        site = settings.SITE
+        location = self.s3.get_bucket_location(Bucket=settings.BUCKET_NAME)['LocationConstraint']
+        logger.debug(location)
+        url = "https://s3-%s.amazonaws.com/%s/%s" % (location, settings.BUCKET_NAME, resource_path)
+        logger.debug(url)
+        return resource_path, url_fix(urljoin(site, url))
```

Comment:
Why are you using the site URL here? This is done in the other method because you build the file URL joining the BAE URL (site) with the file path (resource_path)

Here the file url is directly the AWS URL

urljoin of SITE + url in this case is equal to url, as it is a full url, this is why it is working in runtime